features/support/env.js

```diff
@@ -51,8 +51,24 @@ module.exports = function () {
         } else {
             this.TERMSIGNAL = 'SIGTERM';
             this.EXE = '';
-            // TODO autodetect if this was build with shared or static libraries
-            this.LIB = process.env.BUILD_SHARED_LIBS && '.so' || '.a';
+
+            // if there is libosrm.so/.a we set the .so/.a suffix
+            const dot_so = util.format('%s/libosrm%s', this.BIN_PATH, '.so');
+            const dot_a = util.format('%s/libosrm%s', this.BIN_PATH, '.a');
+
+            // heuristically detect .so/.a suffix, fallback to .a as it's the default
+            this.LIB = '.a';
+
+            try {
+                fs.accessSync(dot_a, fs.F_OK);
+                this.LIB = '.a';
+            } catch(e) { /*nop*/ }
+
+            try {
+                fs.accessSync(dot_so, fs.F_OK);
+                this.LIB = '.so';
+            } catch(e) { /*nop*/ }
+
```

Comment:
wouldn't it be better if this would fail if neither `.a` nor `.so` are detected?

`if (!this.LIB) { throw 'could not detect neither .so nor .a'; }`